build.rs

```diff
@@ -0,0 +1,12 @@
+#![deny(warnings)]
+
+use std::env;
+
+fn main() {
+    let target = env::var("TARGET").expect("TARGET was not set");
+    if target.contains("uwp") {
+        println!("cargo:rustc-link-lib=bcrypt");
+    } else if target.contains("apple-ios") {
+        println!("cargo:rustc-link-lib=framework=Security");
```

Comment:
I am not sure if we need this bit for iOS, but to be safe I've followed the `libstd` [example](https://github.com/rust-lang/rust/blob/master/src/libstd/build.rs).

Reply:
If we're following `libstd`, shouldn't we also add `println!("cargo:rustc-link-lib=advapi32");` for `windows` targets? That's [where `RltGenRandom` comes from](https://docs.microsoft.com/en-us/windows/win32/api/ntsecapi/nf-ntsecapi-rtlgenrandom).

Reply:
Hm, you are right. But I wonder why does it work without explicit linking...

Reply:
It's not need for building the library, and I would guess it's because any Rust Windows binary has some other dep that links `advapi32`, as it's the base DLL for most Windows system calls.

So this is more a exercise in being cautious.

Reply:
To be safe linking of `advapi32` is added.